### PR TITLE
(tests) Log `transform-` instead of `proposal-` in `preset-env`

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

The main reason for this PR is to reduce the output difference between Babel 8 and Babel 7, which is quite annoying (I noticed it while reviewing https://github.com/babel/babel/pull/15068)

As mentioned in https://github.com/babel/babel/pull/14976#discussion_r979299470, this change is only about how we present the log and I don't think it counts as a breaking change.

Users can already use options like `"exclude": ["transform-class-static-blocks"]` (even if technically the npm package name uses `-proposal-`), so it's ok to also use it in the debug logs. Most users should only use the preset and not the individual packages anyway, and this hides from them the distinction between "transform" and "proposal that become Stage 4 since when the plugin was released", which is just historical and doesn't matter in practice.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15473"></a>

### CODE REVIEW

**Review by JLHwung — COMMENTED** [review]
I think changing logs here will be confusing to end users. Take `proposal-optional-chaining` for example:
- We haven't published `@babel/plugin-transform-optional-chaining` yet
- There aren't docs for `@babel/plugin-transform-optional-chaining` on our website, and we don't have redirection rule to the `proposal-*` transform

So users will have to mentally match _some_ `transform-*` debug output to the exact `proposal-*` one. They probably won't know which `transform-*` is actually `proposal-*` until when they hit a npm 404 error, which seems more like a Babel bug to them than a polish to us...

As for fixture changes, I think we may have to duplicate all `debug` fixtures to babel 8.

Can we start the renaming process gradually? We can publish `transform-*` in the next minor and make `proposal-*` an alias to the `transform-*` plugins, then we archive the `proposal-*` plugins and have `preset-env` directly using `transform-*` plugins, so the debug log is consistent.

**Comment by nicolo-ribaudo** [comment]
That plan sounds good 👍 Wdyt if I update this PR to only unify the fixtures (without actually changing what our users see), while working on the rename in parallel? I'd like to avoid duplicating the fixtures, because in practice it means that we have to run the tests twice _every time_ we want to update snapshots (which is easy to forget, until you see the error on CI).

File: packages/babel-preset-env/test/index.skip-bundled.js [hunk 1]
```diff
@@ -309,4 +313,40 @@ describe("babel-preset-env", () => {
...
     expect(arrAvailablePlugins).toEqual(expect.arrayContaining(arrCompatData));
   });
+
+  describe("debug", () => {
+    let log;
+    beforeEach(() => {
+      log = jest.spyOn(console, "log").mockImplementation(() => {});
+    });
+    afterEach(() => {
+      log.restoreMock();
```
Comment (nicolo-ribaudo) [inline packages/babel-preset-env/test/index.skip-bundled.js:323]:
```suggestion
      log.mockRestore();
```

File: packages/babel-helper-transform-fixture-test-runner/src/index.ts [hunk 2]
```diff
@@ -441,6 +441,13 @@ function normalizeOutput(code: string, normalizePathSeparator?: boolean) {
       );
     }
   }
+
+  // In Babel 8, preset-env logs transform- instead of proposal-. Manually rewrite
+  // the output logs so that we don't have to duplicate all the debug fixtures for
+  // the two different Babel versions.
+  // TODO(Babel 8): Remove this.
+  result = result.replace(/(\s+)proposal-/gm, "$1transform-");
```
Comment (JLHwung) [inline packages/babel-helper-transform-fixture-test-runner/src/index.ts:449]: The `normalizeOutput` is also applied to the transformed code. Can we make a new flag, e.g. `normalizePresetEnvDebug` like we did for `normalizePathSeparator`? Then we can enable this flag only when we are validating logs, preferably from `preset-env` only. So the fixture runner won't accidentally replace `proposal-` to `transform-` out of the preset-env scope.

**Review by JLHwung — APPROVED** [review]

**Review by liuxingbaoyu — APPROVED** [review]

**Pull request merged**